li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessResult.java

```diff
@@ -46,14 +56,14 @@ void recordException(TopicPartition tp, long offset, RuntimeException e) {
     _exceptions.add(new RecordProcessingException(tp, offset, e));
     // The resume offset is the error offset + 1. i.e. if user ignore the exception thrown and poll again, the resuming
     // offset should be this one.
-    _resumeOffsets.putIfAbsent(tp, offset + 1);
+    _offsetPair.putIfAbsent(tp, new OffsetPair(offset, offset + 1));
   }
 
   public void clearRecords() {
     _processedRecords = null;
   }
 
-  boolean hasError(TopicPartition tp) {
+  public boolean hasError(TopicPartition tp) {
     return resumeOffsets().containsKey(tp);
```

Comment:
This is kind of expensive as resumeOffsets() constructs a new map each time this check is performed.

Reply:
```
public Map<TopicPartition, OffsetPair> offsets() {
  return Collections.unmodifiableMap(_offsetPair);
}
```
Changed the definition to return an immutable map with the OffsetPair.